### PR TITLE
[OS-728] Delegate WPA supplicant startup to system services

### PR DESCRIPTION
**Pull request by skarbat**
This PR offloads the WPA supplicant startup during bootup to the system services - udhcpcd.
This effectively disables `kano-connect`, and switches to using the original `wpa_supplicant` configuration file.

This PR obsoletes https://github.com/KanoComputing/kano-settings/pull/528
This PR needs this changeset: https://github.com/KanoComputing/kano-toolset/pull/303

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,9 @@
+kano-settings (4.2.1-0) unstable; urgency=low
+
+  * Fixed wireless connection on boot for Kano 4.2.0 release
```
Comment (radujipa) [inline debian/changelog]: I'd rewrite this as `Fixed race condition at boot preventing Wi-Fi from reconnecting to existing network`

File: debian/kano-connect.preinst [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# preinst
+#
+# Copyright (C) 2018 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
+#
+
+case "$1" in
+    install|upgrade)
```
Comment (radujipa) [inline debian/kano-connect.preinst]: I'd suggest 3 tests here with this new script:

 - On a fresh img build, the fix should work
 - From an older version upgrading to this version, the fix should world
 - Upgrading to a newer version of kano-settings and kano-connect, the fix should still work

File: debian/kano-connect.preinst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        #
+        # Tell APT that the wpa_supplicant.conf file is now handled by kano-connect until uninstallation (see postrm)
+        #
+        # https://askubuntu.com/questions/116768/providing-a-customized-config-file-for-another-package
+        # https://wiki.debian.org/Adding%20and%20removing%20diversions
+        #
+        dpkg-divert --add --package kano-connect --rename --divert \
+            /etc/wpa_supplicant/wpa_supplicant.conf.disabled /etc/wpa_supplicant/wpa_supplicant.conf
```
[the remark this answers is not in the document]
Comment (skarbat) [reply debian/kano-connect.preinst]: Tested on a fresh image, think we should be fine. multiple divert additions should be harmless.. There's the question as to why it belongs to kano-toolset, but that's a different story.

```
root@KXBN001:/# /root/multiple-diverts.sh 
Leaving 'diversion of /etc/wpa_supplicant/wpa_supplicant.conf to /etc/wpa_supplicant/wpa_supplicant.conf.disabled by kano-connect'
root@KXBN001:/# /root/multiple-diverts.sh 
Leaving 'diversion of /etc/wpa_supplicant/wpa_supplicant.conf to /etc/wpa_supplicant/wpa_supplicant.conf.disabled by kano-connect'
root@KXBN001:/# /root/multiple-diverts.sh 
Leaving 'diversion of /etc/wpa_supplicant/wpa_supplicant.conf to /etc/wpa_supplicant/wpa_supplicant.conf.disabled by kano-connect'
root@KXBN001:/# dpkg -S /etc/wpa_supplicant/wpa_supplicant.conf.d
wpa_supplicant.conf.disabled  wpa_supplicant.conf.dpkg-new  
root@KXBN001:/# dpkg -S /etc/wpa_supplicant/wpa_supplicant.conf  
diversion by kano-connect from: /etc/wpa_supplicant/wpa_supplicant.conf
diversion by kano-connect to: /etc/wpa_supplicant/wpa_supplicant.conf.disabled
kano-toolset: /etc/wpa_supplicant/wpa_supplicant.conf
```

File: etc/rc.network [hunk 4]
```diff
@@ -9,10 +9,7 @@
...
-device_type="wlan"
-wlan_device=`grep --max-count=1 $device_type /proc/net/dev | \
-             sed -r 's/^.*(wlan[[:digit:]]).*$/\1/'`
-if [ ! "$wlan_device" = "" ]; then
-    python /usr/bin/kano-connect -c $wlan_device &
-fi
+# Automatic wireless connection on boot is now performed by the standard
+# system via the supplicant daemon - see dhcpcd systemd service
+
+exit 0
```
[the remark this answers is not in the document]
Comment (tombettany) [reply etc/rc.network]: Need to make sure that this file actually gets deleted - we've been burned before by updated conffiles in `/etc` sticking around even though they've been deleted.

File: debian/kano-connect.preinst (hunk 3 above)
[the remark this answers is not in the document]
Comment (tombettany) [reply debian/kano-connect.preinst]: Okay, fair enough.

File: etc/rc.network (hunk 4 above)
[the remark this answers is not in the document]
Comment (skarbat) [reply etc/rc.network]: True! Done

**Pull request merged**